pkgs/tools/misc/opentelemetry-collector/contrib.nix

```
@@ -41,16 +42,21 @@ buildGoModule rec {
 
   preCheck = "export CGO_ENABLED=1";
 
-  # This test fails on darwin for mysterious reasons.
-  checkFlags = lib.optionals stdenv.isDarwin
-    [ "-skip" "TestDefaultExtensions/memory_ballast" ];
+  checkFlags = [
+    # collectd test tries and fails to bind to port 8081
+    # mezo test tries to reach external service
+    # memory_ballast test fails on darwin for mysterious reasons.
+    "-skip=TestDefaultReceivers/collectd|TestDefaultExporters/mezmo${if stdenv.isDarwin then "|TestDefaultExtensions/memory_ballast" else ""}"
+  ];
```

```suggestion
  checkFlags = [
    # collectd test tries and fails to bind to port 8081
    # mezmo test tries to reach external service
    "-skip=TestDefaultReceivers/collectd|TestDefaultExporters/mezmo"
  ] ++ lib.optionals stdenv.isDarwin [
    # memory_ballast test fails on darwin for mysterious reasons
    "-skip=TestDefaultExtensions/memory_ballast"
  ];
```

Different formatting to be a bit more readable and fixed a spelling issue in the comments

---

Unfortunately you can't pass the -skip flag multiple times, it overwrites previous calls.